### PR TITLE
[perf] faster docker image builds

### PR DESCRIPTION
**Pull request by das7pad**
### Description

For https://github.com/overleaf/issues/issues/3225

I need to build a couple more server ce images while experimenting with the configs. With docker layer caching and a few other tweaks I can cut the build time from ~30-40min to 17min.
This PR is setting up docker layer caching and uses `npm ci` instead of `npm install`.
All but web use `npm ci --only=production` which omits installing dev-dependencies. We've got linting setup for blocking the use of dev-dependencies in app/ files. Web needs all dev-dependencies for running webpack.

Before: 35min: https://console.cloud.google.com/cloud-build/builds;region=global/1d36ce6e-72a4-49a2-ba56-c7c2b1fb24cd?project=overleaf-ops
After: 17min: https://console.cloud.google.com/cloud-build/builds;region=global/6c124303-8b0f-490b-aacf-d525ec588276?project=overleaf-ops

#### Related Issues / PRs

For https://github.com/overleaf/issues/issues/3225

#### Potential Impact

Low.

### CODE REVIEW

File: bin/install-services [hunk 1]
```diff
@@ -8,7 +8,15 @@ grep 'name:' config/services.js | \
 	do
 		pushd $service
 		echo "Installing service $service"
-		npm install --quiet
+		case $service in
+			web)
+				# install webpack and friends from dev-dependencies.
+				npm ci
```
Comment (mserranom) [inline bin/install-services:14]: Should we keep using `--quiet` for manageable logs?

**Review by mserranom — APPROVED** [review]
Neat!  Always thought docker would be using _any_ image in the system for caching!

Added a minor comment up to your consideration.

File: bin/install-services (hunk 1 above)
Comment (das7pad) [reply bin/install-services:14]: I tried `npm ci --quiet` locally for the notifications service and it did not make any difference :shrug:

Comment (mserranom) [reply bin/install-services:14]: Weird, according to npm v7 `--quiet` is still equivalent to --loglevel warn` https://docs.npmjs.com/cli/v7/using-npm/config

**Pull request merged**